Gulpfile.js

```diff
@@ -205,19 +304,80 @@ const standaloneBundle = [
   },
 ];
 
+gulp.task("generate-type-helpers", () => {
+  fancyLog("Generating @babel/types dynamic functions");
+  return Promise.all(
+    ["asserts", "builders", "constants", "validators"].map(helperKind =>
+      generateTypeHelpers(helperKind)
+    )
+  );
+});
+
+gulp.task("build-flow-typings", () => buildFlowTypings());
+gulp.task("build-typescript-typings", () => buildTypeScriptTypings());
+gulp.task(
+  "build-typings",
+  gulp.parallel("build-flow-typings", "build-typescript-typings")
+);
+
+gulp.task("generate-standalone", () => generateStandalone());
+
 gulp.task("build-rollup", () => buildRollup(libBundles));
-gulp.task("build-babel-standalone", () => buildRollup(standaloneBundle));
+gulp.task("rollup-babel-standalone", () => buildRollup(standaloneBundle));
+gulp.task(
+  "build-babel-standalone",
+  gulp.series("generate-standalone", "rollup-babel-standalone")
+);
 
 gulp.task("build-babel", () => buildBabel(/* exclude */ libBundles));
-gulp.task("build", gulp.parallel("build-rollup", "build-babel"));
+
+gulp.task(
+  "build",
+  gulp.series(
+    gulp.parallel("build-rollup", "build-babel"),
+    gulp.parallel(
+      "generate-standalone",
+      "build-typings",
+      gulp.series(
+        "generate-type-helpers",
+        // rebuild @babel/types since type-helpers may be changed
+        "build-babel"
+      )
+    )
+  )
+);
 
 gulp.task("default", gulp.series("build"));
 
 gulp.task("build-no-bundle", () => buildBabel());
 
+gulp.task(
+  "build-dev",
+  gulp.series(
+    "build-no-bundle",
+    gulp.parallel(
+      "generate-standalone",
+      "build-typings",
+      gulp.series(
+        "generate-type-helpers",
+        // rebuild @babel/types since type-helpers may be changed
+        "build-no-bundle"
+      )
+    )
+  )
+);
+
 gulp.task(
   "watch",
-  gulp.series("build-no-bundle", function watch() {
+  gulp.series("build-dev", function watch() {
     gulp.watch(defaultSourcesGlob, gulp.task("build-no-bundle"));
+    gulp.watch(
+      babelStandalonePluginConfigGlob,
+      gulp.task("generate-standalone")
+    );
+    gulp.watch(
+      buildTypingsWatchGlob,
+      gulp.parallel("build-typings", "generate-type-helpers")
```

Comment:
When you edit `packages/babel-types/src/definitions` _or_ `./packages/babel-types/scripts/generators/*.js`, the `@babel/types` helpers and typings will be regenerated automatically. So you don't have to run (and remember to!) `make generate-type-helpers && make build-typings` again.